tests/cupy_tests/indexing_tests/test_insert.py

```diff
@@ -148,3 +148,37 @@ def test_1darray(self, xp, dtype):
         a = testing.shaped_arange(5, xp, dtype)
         val = self._compute_val(xp)
         xp.fill_diagonal(a, val=val, wrap=self.wrap)
+
+
+@testing.parameterize(*testing.product({
+    'n': [2, 4],
+    'ndim': [2, 3],
+}))
```

Comment:
Test invalid values too

Reply:
I think there are no invalid values for this test, any pair of numbers would work fine.

Reply:
negative values for `n` and `ndim`, unidimensional cases, etc, ...
values that should make the routine fail or give an error.
Also corner cases such as 0, dim matrix etc

Reply:
Those cases also don't fail or give any error on NumPy (They return empty tuples and so on.) But yes, we should definitely add those cases in the tests :+1:

Reply:
added those cases!

Reply:
Please make a specific class for invalid tests

Reply:
Please make a specific class for invalid tests

Reply:
Oh, I am sorry, I didn't notice your earlier comment until now. I have separated those cases.

Reply:
can you take a look, please? @emcastillo